packages/contracts-bedrock/contracts/echidna/FuzzOracle.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity 0.8.15;
+
+import { L2OutputOracle } from "../L1/L2OutputOracle.sol";
+import { Types } from "../libraries/Types.sol";
+
+/**
+ * @dev Modified version of `L2OutputOracle` that contains a less restrictive
+ * version `proposeL2Output` for use with echidna.
+ */
+contract ModL2OutputOracle is L2OutputOracle {
+    constructor(
+        uint256 _submissionInterval,
+        uint256 _l2BlockTime,
+        uint256 _startingBlockNumber,
+        uint256 _startingTimestamp,
+        address _proposer,
+        address _challenger
+    ) L2OutputOracle(
+        _submissionInterval,
+        _l2BlockTime,
+        _startingBlockNumber,
+        _startingTimestamp,
+        _proposer,
+        _challenger
+    ) {
+        // ...
+    }
+
+    /**
+     * @dev This function is a modified version of `proposeL2Output` that removes
+     * several checks which limit echidna.
+     *
+     * Because the purpose of this test is to check that the block number of the
+     * proposed outputs must monotonically increase, these checks are safe to remove.
+     */
+    function proposeLessChecks(
+        bytes32 _outputRoot,
+        uint256 _l2BlockNumber
+    ) external {
+        require(
+            _l2BlockNumber == nextBlockNumber(),
+            "L2OutputOracle: block number must be equal to next expected block number"
+        );
+
+        emit OutputProposed(_outputRoot, nextOutputIndex(), block.timestamp, _l2BlockNumber);
+
+        l2Outputs.push(
+            Types.OutputProposal({
+                outputRoot: _outputRoot,
+                timestamp: uint128(block.timestamp),
+                l2BlockNumber: uint128(_l2BlockNumber)
+            })
+        );
+    }
+}
+
+contract EchidnaFuzzL2OutputOracle {
+    // ENV
+    uint256 internal immutable submissionInterval = 1800;
```

Comment:
I never knew that you could assign to an `immutable` outside of the constructor, thought it would have to be `constant`.

Reply:
Ah yeah this works- immutables can either be defined directly or within the constructor. Will swap over to constants to follow the convention used in the rest of the codebase.